Tooltip enhancement

ldong opened the pull request:
Arrow pointer enhancement
1. Add _up_ and _down_ arrow support
2. Add mask
3. Add auto-resize
4. Fix typo `initTootltip` to `initTooltip`

Screenshot:

![tooltip](https://cloud.githubusercontent.com/assets/5491695/18295423/2f9ccf84-7456-11e6-93f3-edef354e76d9.gif)

ldong committed:
Add up and down arrow

Add up and down arrow support

ldong committed:
Arrow pointer enhancement

1. Add up and down arrow support
2. Add mask

Commit:
Arrow pointer enhancement

1. Add up and down arrow support
2. Add mask

svkangal commented on addon/components/ember-tutorial-component.js:
```diff
@@ -167,17 +179,37 @@ export default Ember.Component.extend({
+   * @method didDestroyElement
+   */
+  didDestroyElement() {
+    console.log('resize position and trigger setPosition');
```
Remove console log. User Ember Logger instead

svkangal commented on addon/components/ember-tutorial-component.js:
```diff
@@ -191,27 +223,120 @@ export default Ember.Component.extend({
+   * @method turn off resize listeners
+   */
+  cancelAutoResize() {
+    console.log('cancelAutoResize');
```
Remove console log. User Ember Logger instead

svkangal commented on addon/components/ember-tutorial-component.js:
```diff
@@ -191,27 +223,120 @@ export default Ember.Component.extend({
+  computeXYCord(tooltipElementSelector='.tutorial-component') {
+    let $mainElement = Ember.$(this.get('currentConfig').ele);
+    let $tooltipElement = Ember.$(tooltipElementSelector);
+    const borderWidth = this.getBorderWidth(tooltipElementSelector);
```
Do we need to calculate border width? getBoundingClientRect gives you width and height including the border widths and padding.

svkangal replied:
No worries I see now what it has been used for.

svkangal commented on addon/components/ember-tutorial-component.js:
```diff
@@ -191,27 +223,120 @@ export default Ember.Component.extend({
+    this.set('yCoord', y);
+  },
+
+  didInsertElement() {
```
why do we need this?

svkangal commented on addon/templates/components/ember-tutorial-component.hbs:
```diff
@@ -1,5 +1,6 @@
 {{#ember-wormhole to="ember-tutorial-component-wormhole"}}
-  <div class='tutorial-component {{if isLeftPointer "left" "right"}} {{if hideMessage "hide" "show"}}' style={{tooltipStyle}}>
+<div class='tutorial-component-mask {{if hideMessage "hide" "show"}}'>
```
Lets keep tutorial-component-mask optional and part of config.

svkangal commented on addon/components/ember-tutorial-component.js:
```diff
@@ -167,17 +179,37 @@ export default Ember.Component.extend({
+
+    Ember.$(window).on('resize.tutorial-component', function() {
+      console.log('resize position and trigger setPosition');
+      this.setPosition();
```
Thanks for adding this, wanted to add that in the initial release itself but missed it.

svkangal commented on addon/components/ember-tutorial-component.js:
```diff
@@ -191,27 +223,120 @@ export default Ember.Component.extend({
+    const position = this.get('pointerDirection').toLowerCase();
+    switch(position) {
+      case 'left':
+        x = X + W + borderWidth;
```
lets discuss this logic, have some questions.

svkangal commented on addon/components/ember-tutorial-component.js:
```diff
@@ -191,27 +223,120 @@ export default Ember.Component.extend({
+    }
+  },
+
+  computeXYCord(tooltipElementSelector='.tutorial-component') {
```
Please add some documentation every where. Also if possible add some inline comments explaining the code.

svkangal commented:
Will be great if you can add some tests too and also update the Readme.

Commit:
Made masking declarative as part of configuration, change console log to Ember log

1. Change console.log to Ember Log
2. Add documentation to indicate the usage of up, right, down, left for arrow pointer position
3. Add mask as optional field into config
4. Add mask button in the demo page, application.hbs

Commit:
Update README

ldong committed:
Fix typo

ldong commented:
1. Change console.log to Ember Log
2. Add documentation to indicate the usage of up, right, down, left for arrow pointer position
3. Add mask as optional field into config
4. Add mask button in the demo page, application.hbs
5. Update _README_
